experiments/AMIP/coupler_driver.jl

```diff
@@ -554,38 +619,43 @@ else
     error("turb_flux_partition must be either PartitionedStateFluxes or CombinedStateFluxes")
 end
 
-# 1) coupler combines surface states and calculates rho_sfc using surface and atmos variables
+# 2. coupler updates surface model area fractions
 update_surface_fractions!(cs)
+
+# 3. surface density (`ρ_sfc`): calculated by the coupler by adiabatically extrapolating atmospheric thermal state to the surface.
+# For this, we need to import surface and atmospheric fields. The model sims are then updated with the new surface density.
 import_combined_surface_fields!(cs.fields, cs.model_sims, cs.boundary_space, turbulent_fluxes)
 import_atmos_fields!(cs.fields, cs.model_sims, cs.boundary_space, turbulent_fluxes)
 update_model_sims!(cs.model_sims, cs.fields, turbulent_fluxes)
 
-# 2) each surface component model calculates its own vapor specific humidity (q_sfc)
-# TODO: the q_sfc calculation follows the design of the bucket q_sfc, but it would be neater to abstract this from step!
+# 4. surface vapor specific humidity (`q_sfc`): step surface models with the new surface density to calculate their respective `q_sfc` internally
+## TODO: the q_sfc calculation follows the design of the bucket q_sfc, but it would be neater to abstract this from step! (#331)
 step!(land_sim, Δt_cpl)
 step!(ocean_sim, Δt_cpl)
 step!(ice_sim, Δt_cpl)
 
-# 3) coupler re-imports updated surface fields and calculates turbulent fluxes, while updating atmos sfc_conditions
+# 5. turbulent fluxes: now we have all information needed for calculating the initial turbulent surface fluxes using the combined state
+# or the partitioned state method
 if turbulent_fluxes isa CombinedStateFluxes
-    # calculate fluxes using combined surface states on the atmos grid
+    ## import the new surface properties into the coupler (note the atmos state was also imported in step 3.)
     import_combined_surface_fields!(cs.fields, cs.model_sims, cs.boundary_space, turbulent_fluxes) # i.e. T_sfc, albedo, z0, beta, q_sfc
+    ## calculate turbulent fluxes in surface models and save the weighted average in coupler fields
     combined_turbulent_fluxes!(cs.model_sims, cs.fields, turbulent_fluxes) # this updates the atmos thermo state, sfc_ts
 elseif turbulent_fluxes isa PartitionedStateFluxes
-    # calculate turbulent fluxes in surface models and save the weighted average in coupler fields
+    ## calculate turbulent fluxes in surface models and save the weighted average in coupler fields
```

Comment:
this comment is duplicate of what we have in line 642

Reply:
Ahh good catch!